proposals/infra/INF-0002-root-signature-in-clang.md

```diff
@@ -304,15 +451,196 @@ elimation has completed.
+  - DESCRIPTORS_VOLATILE | DATA_VOLATILE
+  - DESCRIPTORS_VOLATILE | DATA_STATIC_WHILE_SET_AT_EXECUTE
+  - DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS
+  - DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_VOLATILE
+  - DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_STATIC
+  - DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_STATIC_WHILE_SET_AT_EXECUTE
+
+- Valid values for DescriptorRangeFlags on Sampler
+  - 0
+  - DESCRIPTORS_VOLATILE
+  - DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS
+
+- StaticSampler
+  - Valid values for Filter
+    - FILTER_MIN_MAG_MIP_POINT
+    - FILTER_MIN_MAG_POINT_MIP_LINEAR
+    - FILTER_MIN_POINT_MAG_LINEAR_MIP_POINT
+    - FILTER_MIN_POINT_MAG_MIP_LINEAR
+    - FILTER_MIN_LINEAR_MAG_MIP_POINT
+    - FILTER_MIN_LINEAR_MAG_POINT_MIP_LINEAR
+    - FILTER_MIN_MAG_LINEAR_MIP_POINT
+    - FILTER_MIN_MAG_MIP_LINEAR
+    - FILTER_ANISOTROPIC
+    - FILTER_COMPARISON_MIN_MAG_MIP_POINT
+    - FILTER_COMPARISON_MIN_MAG_POINT_MIP_LINEAR
+    - FILTER_COMPARISON_MIN_POINT_MAG_LINEAR_MIP_POINT
+    - FILTER_COMPARISON_MIN_POINT_MAG_MIP_LINEAR
+    - FILTER_COMPARISON_MIN_LINEAR_MAG_MIP_POINT
+    - FILTER_COMPARISON_MIN_LINEAR_MAG_POINT_MIP_LINEAR
+    - FILTER_COMPARISON_MIN_MAG_LINEAR_MIP_POINT
+    - FILTER_COMPARISON_MIN_MAG_MIP_LINEAR
+    - FILTER_COMPARISON_ANISOTROPIC
+    - FILTER_MINIMUM_MIN_MAG_MIP_POINT
+    - FILTER_MINIMUM_MIN_MAG_POINT_MIP_LINEAR
+    - FILTER_MINIMUM_MIN_POINT_MAG_LINEAR_MIP_POINT
+    - FILTER_MINIMUM_MIN_POINT_MAG_MIP_LINEAR
+    - FILTER_MINIMUM_MIN_LINEAR_MAG_MIP_POINT
+    - FILTER_MINIMUM_MIN_LINEAR_MAG_POINT_MIP_LINEAR
+    - FILTER_MINIMUM_MIN_MAG_LINEAR_MIP_POINT
+    - FILTER_MINIMUM_MIN_MAG_MIP_LINEAR
+    - FILTER_MINIMUM_ANISOTROPIC
+    - FILTER_MAXIMUM_MIN_MAG_MIP_POINT
+    - FILTER_MAXIMUM_MIN_MAG_POINT_MIP_LINEAR
+    - FILTER_MAXIMUM_MIN_POINT_MAG_LINEAR_MIP_POINT
+    - FILTER_MAXIMUM_MIN_POINT_MAG_MIP_LINEAR
+    - FILTER_MAXIMUM_MIN_LINEAR_MAG_MIP_POINT
+    - FILTER_MAXIMUM_MIN_LINEAR_MAG_POINT_MIP_LINEAR
+    - FILTER_MAXIMUM_MIN_MAG_LINEAR_MIP_POINT
+    - FILTER_MAXIMUM_MIN_MAG_MIP_LINEAR
+    - FILTER_MAXIMUM_ANISOTROPIC
+
+  - Valid values for TextureAddress
+    - TEXTURE_ADDRESS_WRAP 
+    - TEXTURE_ADDRESS_MIRROR  
+    - TEXTURE_ADDRESS_CLAMP 
+    - TEXTURE_ADDRESS_BORDER 
+    - TEXTURE_ADDRESS_MIRROR_ONCE
+
+  - Valid values for ComparisonFunc
+    - 0
+    - COMPARISON_NEVER 
+    - COMPARISON_LESS 
+    - COMPARISON_EQUAL 
+    - COMPARISON_LESS_EQUAL 
+    - COMPARISON_GREATER 
+    - COMPARISON_NOT_EQUAL 
+    - COMPARISON_GREATER_EQUAL 
+    - COMPARISON_ALWAYS
+
+  - Valid values for StaticBorderColor
+    - STATIC_BORDER_COLOR_TRANSPARENT_BLACK 
+    - STATIC_BORDER_COLOR_OPAQUE_BLACK 
+    - STATIC_BORDER_COLOR_OPAQUE_WHITE
+
+  - Comparison filter must have ComparisonFunc not equal to 0.
+      ```
+      When the Filter of a StaticSampler is FILTER_COMPARISON*,
+      the ComparisonFunc cannot be 0.
+      ```
```

Comment:
Should the `RootFlag`s also be validated?

Reply:
Updated.